Sources/Sentry/SentryHub.m

```diff
@@ -166,10 +166,17 @@ - (SentryClient *_Nullable)getClient {
 }
 
 - (SentryScope *)getScope {
-    if (self.scope == nil) {
-        self.scope = [[SentryScope alloc] init];
+    @synchronized (self) {
+        if (self.scope == nil) {
+            SentryClient *client = [self getClient];
+            if (nil != client) {
+                self.scope = [[SentryScope alloc] initWithMaxBreadcrumbs:client.options.maxBreadcrumbs];
```

Comment:
would it make sense to pass the whole `options` instead of only `maxBreadcrumbs`? on Android, we also use `BeforeBreadcrumbCallback` that also comes from the `options`.
ps: we use the `options` as a kind of `service locator`

Reply:
Would work of course, I just didn't want that the scope needs to care about too many things. It shouldn't know about `SentryOptions` (yet) only because of int (maxBreadcrumbs).

I think we should [KISS](https://en.wikipedia.org/wiki/KISS_principle) this.

Reply:
definitely agree we should KISS it.
we've changed it because of `BeforeBreadcrumbCallback` and `ILogger`